conftest.py

```diff
@@ -211,3 +250,10 @@ def pytest_generate_tests(metafunc):
         if not vm_lists:
             vm_lists = [None] # no --vms parameter does not mean skip the test, for us, it means use the default
         metafunc.parametrize("vm_refs", vm_lists, indirect=True, scope="module")
+    if "sr_deviceconfig" in metafunc.fixturenames:
+        configs = metafunc.config.getoption("sr_deviceconfig")
+        if not configs:
+            # No --sr-deviceconfig parameter doesn't mean skip the test.
+            # For us it means use the defaults.
+            configs = [None]
```

Comment:
Why not  `[]`?

Reply:
Because of the way fixtures work. To pytest, an empty list means "no fixtures, skip the tests that require it", so I provide a non-empty list with a fixture initialized to value None and then I handle that case in the fixture function.

Reply:
And why it is a list and not a single value: because you can provide several values for the same fixture and then pytest will execute the tests for each fixture.

Reply:
Thanks for the info :)